test/ssltest_old.c

```diff
@@ -1382,11 +1382,28 @@ int main(int argc, char *argv[])
         goto end;
 
     if (cipher != NULL) {
-        if (!SSL_CTX_set_cipher_list(c_ctx, cipher)
-            || !SSL_CTX_set_cipher_list(s_ctx, cipher)
-            || !SSL_CTX_set_cipher_list(s_ctx2, cipher)) {
-            ERR_print_errors(bio_err);
-            goto end;
+        if (strcmp(cipher, "") == 0) {
+            if (!SSL_CTX_set_cipher_list(c_ctx, cipher)
+                    && !SSL_CTX_set_cipher_list(s_ctx, cipher)
+                    && !SSL_CTX_set_cipher_list(s_ctx2, cipher)) {
+                if (ERR_GET_REASON(ERR_peek_error()) == SSL_R_NO_CIPHER_MATCH) {
```

Comment:
Really we should do this after each call to `SSL_CTX_set_cipher_list` above, i.e. each one of them should create an error - but we only check once at the end.